pkg/codegen/nodejs/gen_program_utils.go

```diff
@@ -18,6 +18,14 @@ func getHelperMethodIfNeeded(functionName string, indent string) (string, bool)
 			`%sfunction notImplemented(message: string) {
 %s    throw new Error(message);
 %s}`, indent, indent, indent), true
+	case "singleOrNone":
+		return fmt.Sprintf(
+			`%sfunction singleOrNone<T>(elements: pulumi.Input<T>[]): pulumi.Input<T> | undefined {
+%s    if (elements.length == 1) {
```

Comment:
I wonder if we should error for length>1 rather than undef?

Reply:
I thought of that but chose a more lenient implementation

Reply:
I think given we'll be using this to mostly translate cases of "MaxItemsOne" from lists to scalars it would be more correct for this to error if length>1.

Reply:
Done ✅